app/controllers/pages_controller.rb

```diff
@@ -83,8 +83,8 @@ def badge
   end
 
   def report_abuse
-    referer = URL.sanitized_referer(request.referer)
-    reported_url = params[:reported_url] || params[:url] || referer
+    reported_url = params[:reported_url] || params[:url] || referer.presence
```

Comment:
The referer needs to be accessed via the request object (in the past the plain `referer` was the variable that no longer exists). I think this one is the reason a spec is failing.

Reply:
I went a little overboard with the deletion. 🙃